providers/certificate.rb

```diff
@@ -0,0 +1,160 @@
+		hash = getHashFromFile(file)
+	else
+		# make sure we have no spaces in the hash string
+		hash = @new_resource.source.gsub(/\s/, '')
+	end
+	
+	if isCertInStore?(hash)
+		converge_by("Adding to ACL of #{hash}") do
+			updateAcl!(hash)
+		end
+	else
+		Chef::Log.debug("#{@new_resource.source} does not exist - nothing to do")
+	end
+end
+
+action :delete do
+	# on delete the source should id the cert (subject, hash, serial number etc.)
+	if isCertInStore?(@new_resource.source)
+		converge_by("Deleting #{ @current_resource.source }") do
+			user = " -user" if @new_resource.user_store
+			cmd = shell_out!("#{@command}#{user} -delstore #{@new_resource.store_name} \"#{@current_resource.source}\"")
+		end
+	else
+		Chef::Log.debug("#{@new_resource.source} does not exist - nothing to do")
+	end
+end
+
+def load_current_resource
+	@current_resource = Chef::Resource::WindowsCertificate.new(@new_resource.name)
+	@current_resource.source(@new_resource.source)
+	@current_resource.pfx_password(@new_resource.pfx_password)
+	@current_resource.private_key_acl(@new_resource.private_key_acl)
+	@current_resource.store_name(@new_resource.store_name)
+	@current_resource.user_store(@new_resource.user_store)
+
+	@command = locate_sysnative_cmd("certutil.exe")
+end
+
+private
+def getHashFromFile(file)
+	if file.downcase.end_with?('pfx') && (@new_resource.pfx_password.nil? || @new_resource.pfx_password.empty?)
+		raise "No password given for PFX file"
+	end
+	
+	password = " -p #{@new_resource.pfx_password}" if file.downcase.end_with?('pfx')
+	cmd = shell_out("#{@command}#{password} \"#{file}\"")
+	Chef::Log.debug "certutil reports: #{cmd.stdout}"
+	
+	# extract values from returned text
+	if cmd.exitstatus == 0
+		m = cmd.stdout.scan(/Cert Hash\(sha1\): (.+)/)
+		if m.length > 0
+			# remove spaces from the hash
+			m[m.length - 1][0].gsub(/\s/, '')
+		else
+			raise "Couldn't find hash in output : #{cmd.stdout}"
+		end
+	else
+		raise "certutil returned error #{cmd.exitstatus} : #{cmd.stderr} #{cmd.stdout}"
+	end
+end
+
+def isCertInStore?(certHash)
+	user = " -user" if @new_resource.user_store
+	cmd = shell_out("#{@command}#{user} -store #{@new_resource.store_name} \"#{certHash}\"")
+	Chef::Log.debug "certutil reports: #{cmd.stdout}"
+
+	cmd.exitstatus == 0
+end
+
+def addCertToStore!(file)
+	user = " -user" if @new_resource.user_store
+	if file.downcase.end_with?('pfx')
+		shell_out!("#{@command}#{user} -p #{@new_resource.pfx_password} -importpfx \"#{file}\"")
+	else
+		shell_out!("#{@command}#{user} -addstore #{@new_resource.store_name} \"#{file}\"")
+	end
+end
+
+def updateAcl!(certHash)
+	if (@new_resource.private_key_acl.nil? || @new_resource.private_key_acl.length == 0)
+		return
+	end
+	
+	raise "ACL can only be set on local machine certificates" if @new_resource.user_store
+	
+	# this PS came from http://blogs.technet.com/b/operationsguy/archive/2010/11/29/provide-access-to-private-keys-commandline-vs-powershell.aspx
```

Comment:
Put this in a powershell_script code block so that you don't have to append things to a string,